src/cattr/gen.py

```diff
@@ -109,7 +109,9 @@ def generate_mapping(cl: Type, old_mapping):
     return cls(**mapping)
 
 
-def make_dict_structure_fn(cl: Type, converter, **kwargs):
+def make_dict_structure_fn(
+    cl: Type, converter, forbid_extra_fields: bool = False, **kwargs
```

Comment:
The argument at this layer should be a little obfuscated, since it might shadow an attribute name you're trying to override (in **kwargs). I'd change it to `_cattrs_forbid_extra_fields`. Note that we can add this flag on the `GenConverter` level with a better name.

Reply:
Ah, I wondered about that.  Will do.